osbs/utils/yaml.py

```diff
@@ -72,7 +78,20 @@ def read_yaml(yaml_data, schema, package=None):
             if path.startswith('.'):
                 path = path[1:]
 
-            logger.error('validation error (%s): %s', path or 'at top level', error.message)
-        raise
+            # get all context messages without duplicates caused by 'anyOf' validator
+            error_contexts = set()
+            for context in error.context:
+                error_contexts.add(context.message)
+
+            error_message = "{}: validating '{}' has failed ({})".format(
+                            path or 'at top level', error.validator,
+                            ", ".join(error_contexts) or error.message)
+
+            if error is exc:
```

Comment:
Are you sure that this is safe?

Reply:
Obviously caught error will always be in validator.iter_errors(data), so it should be safe

Reply:
Obviously is strong saying. Provide unittest for that.

Just my poor testing:

```python
In [1]: import jsonschema                                                                                                                                                                                                                      

In [2]: %paste                                                                                                                                                                                                                                 
schema1 = {
    "items": {
        "anyOf": [
            {"type": "string", "maxLength": 2},
            {"type": "integer", "minimum": 5}
        ]
    },
    "minItems": 4
}
## -- End pasted text --

In [3]: instance1 = [{}, 3, "foo"]                                                                                                                                                                                                             

In [4]: exc = None 
   ...: try: 
   ...:     jsonschema.validate(instance, schema) 
   ...: except Exception as e: 
   ...:     exc = e 
   ...:                                                                                                                                                                                                                                        

In [5]: exc                                                                                                                                                                                                                                    
Out[5]: NameError("name 'instance' is not defined")

In [6]: id(exc)                                                                                                                                                                                                                                
Out[6]: 139736100876944

In [8]: validator = jsonschema.Draft4Validator(schema=schema1)                                                                                                                                                                                 

In [10]: errors_iter = list(validator.iter_errors(instance1))                                                                                                                                                                                  

In [11]: id_errros = [id(x) for x in errors_iter]                                                                                                                                                                                              

In [12]: id_errros                                                                                                                                                                                                                             
Out[12]: [139736382952736, 139736382953968, 139736382955200, 139736382957280]

In [13]: exc in errors_iter                                                                                                                                                                                                                    
Out[13]: False

In [14]: id(exc) in id_errros                                                                                                                                                                                                                  
Out[14]: False
```